// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTPeerRequest";
option optimize_for = LITE_RUNTIME;

import public "Bailment.proto";
import public "ConnectionInfo.proto";
import public "Faucet.proto";
import public "OutBailment.proto";
import public "PeerEnums.proto";
import public "PendingBailment.proto";
import public "Signature.proto";
import public "StoreSecret.proto";
import public "VerificationOffer.proto";

message PeerRequest {
    optional uint32 version = 1;
    optional string id = 2;
    optional string initiator = 3;
    optional string recipient = 4;
    optional PeerRequestType type = 5;
    optional string cookie = 6;
    optional Signature signature = 7;
    oneof request {
        Bailment bailment = 8;
        OutBailment outbailment = 9;
        PendingBailment pendingbailment = 10;
        ConnectionInfo connectioninfo = 11;
        StoreSecret storesecret = 12;
        VerificationOffer verificationoffer = 14;
        Faucet faucet = 15;
    }
    optional string server = 13;
}
